backend/molgenis-emx2-tasks/src/main/java/org/molgenis/emx2/tasks/ScriptTask.java

```diff
@@ -0,0 +1,207 @@
+package org.molgenis.emx2.tasks;
+
+import static org.apache.commons.text.StringEscapeUtils.escapeXSI;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Comparator;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.molgenis.emx2.MolgenisException;
+import org.molgenis.emx2.Row;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ScriptTask extends Task<ScriptTask> {
+  private static Logger logger = LoggerFactory.getLogger(ScriptTask.class);
+  private String name;
+  private String script;
+  private String outputFileExtension;
+  private String parameters;
+  private String token;
+  private String dependencies;
+  private Process process;
+  private byte[] output;
+
+  public ScriptTask(String name) {
+    super("Executing script '" + name + "'");
+    this.name = name;
+  }
+
+  public ScriptTask(Row scriptMetadata) {
+    this(scriptMetadata.getString("name"));
+    this.script(scriptMetadata.getString("script"))
+        .outputFileExtension(scriptMetadata.getString("outputFileExtension"))
+        .dependencies(scriptMetadata.getString("dependencies"))
+        .cronExpression(scriptMetadata.getString("cron"));
+  }
+
+  @Override
+  public void run() {
+    if (getStatus().equals(TaskStatus.ERROR)) {
+      // when already errored before start, e.g. on unschedule
+      return;
+    }
+    if (script == null) {
+      this.setError("Script is required");
+      return;
+    }
+
+    this.start();
+    // temporary files with script and to collect output
+    Path tempDir = null;
+    try {
+      try {
+        // create tmp directory
+        tempDir = Files.createTempDirectory("python"); // NOSONAR
```

Comment:
NOSONAR ? maybe its trying to tell us something

Reply:
SONAR is very picky on anything that looks like 'files', basically always asking you to double check. This is a way to tell SONAR we did this on purpose. Otherwise, we would need to check a box in sonar itself